timemachine/md/minimizer.py

```diff
@@ -277,3 +278,83 @@ def equilibrate_host(
     ctxt.multiple_steps(np.linspace(0.0, 0.0, n_steps))
 
     return ctxt.get_x_t(), ctxt.get_box()
+
+
+def get_val_and_grad_fn(impls, box, lamb):
+    """
+    Convert impls, box, lamb into a function that only takes in coords.
+    """
+
+    def val_and_grad_fn(coords):
+        nrgs = []
+        grads = []
+        for impl in impls:
+            g, _, u = impl.execute(coords, box, lamb)
+            nrgs.append(u)
+            grads.append(g)
+        return np.sum(nrgs, axis=0), np.sum(grads, axis=0)
```

Comment:
Nit: performance -- should this use SummedPotential?

Reply:
good call - let me update.